lib/authz/project_purge/project_purge_workflow.go

```diff
@@ -0,0 +1,149 @@
+package project_purge
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/chef/automate/lib/cereal"
+)
+
+const (
+	startPurgeProjectTaskName = "StartPurgeProjects"
```

Comment:
a better name might be `PurgeProjectForDomain`. There's no starting here. We used start in the project tagging stuff because the task called ES to start a task that would run asynchronously and we had another task to poll it for completion.